src/Swarm/Game/Universe.hs

```diff
@@ -0,0 +1,63 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE TemplateHaskell #-}
+
+-- |
+-- SPDX-License-Identifier: BSD-3-Clause
+module Swarm.Game.Universe where
+
+import Control.Lens (makeLenses, view)
+import Data.Function (on)
+import Data.Int (Int32)
+import Data.Text (Text)
+import Data.Yaml (FromJSON, ToJSON, Value (Object), parseJSON, withText, (.:))
+import GHC.Generics (Generic)
+import Linear (V2 (..))
+import Swarm.Game.Location
+
+data SubworldName = DefaultRootSubworld | SubworldName Text
+  deriving (Show, Eq, Ord, Generic, ToJSON)
```

Comment:
This `DefaultRootSubworld` constructor worked out nicely.  Previously I was using an "in-band" constant name `"root"` as a default (i.e. when a subworld name is not provided in the scenario file), which became problematic.